pkg/analyze/selector/engine/iterator_test.go

```diff
@@ -0,0 +1,180 @@
+package engine
+
+import (
+	"cmp"
+	"fmt"
+	"maps"
+	"reflect"
+	"slices"
+	"testing"
+)
+
+func NewPredicateUnchecked(label string, f any) *Predicate {
+	predicate, err := NewPredicate(label, f)
+	if err != nil {
+		panic(err)
+	}
+
+	return predicate
+}
+
+func NewRelationUnchecked(lhs, rhs string, f any) Relation {
+	relation, err := NewRelation(lhs, rhs, f)
+	if err != nil {
+		panic(err)
+	}
+
+	return relation
+}
+
+func MakeRelations(
+	types map[string]reflect.Type,
+	relations []Relation,
+) Relations {
+	result := NewRelations()
+
+	for name, typ := range types {
+		if !result.Add(name, typ) {
+			panic("Invalid field")
+		}
+	}
+
+	for _, relation := range relations {
+		if !result.Relate(relation) {
+			panic("Invalid relation")
+		}
+	}
+
+	return result
+}
+
+func CompareMaps(x, y map[string]any) int {
+	result := cmp.Compare(len(x), len(y))
+	if result != 0 {
+		return result
+	}
+
+	keys := slices.Sorted(maps.Keys(x))
+	otherKeys := slices.Sorted(maps.Keys(y))
+	result = slices.Compare(keys, otherKeys)
+
+	if result != 0 {
+		return result
+	}
+
+	for _, key := range keys {
+		result = cmp.Compare(
+			fmt.Sprintf("%+v", x[key]),
+			fmt.Sprintf("%+v", y[key]),
+		)
```

Comment:
This seems a little suspicious. [reflect.DeepEquals](https://pkg.go.dev/reflect#DeepEqual) may be a good replacement, I'm not sure if substituting this with DeepEquals produces equivalent code, though.